### ISSUE TITLE
Version 0.2.x seems to be broken

### ISSUE THREAD

**Issue by werne2j** [IBM/argocd-vault-plugin#40]
### **Workaround is to continue to use version 0.1.0**

`failed to unmarshal manifest: error converting YAML to JSON: yaml: mapping values are not allowed in this context`

There seems to be an issue with some yaml marshaling somewhere

**Comment by jkayani** [comment]
I think this version is broken specifically in cases where the `argocd-repo-server` doesn't run as `argocd`. In our situation, this is in Openshift 4 using the random uid assigned via the `restricted` SCC. 

On a vanilla cluster using `argocd` user in the repo-server pod, this is the result after running the plugin:
```
argocd@argocd-repo-server-7dc6c547c5-xswgw:~$ ls -lra .
total 36
drwxr-xr-x 7 argocd argocd 4096 Jan 13 21:08 ibm-object-storage-plugin
drwx------ 2 argocd argocd 4096 Jan 25 17:59 .ssh
drwxr-xr-x 3 argocd argocd 4096 Jan 13 21:08 .local
drwxr-xr-x 3 argocd argocd 4096 Jan 13 21:08 .config
drwxr-xr-x 3 argocd argocd 4096 Jan 13 21:08 .cache
drwxr-xr-x 2 argocd argocd 4096 Jan 25 17:59 .avp
drwxr-xr-x 1 root   root   4096 Jan 10 05:32 ..
drwxrwxr-x 1 argocd root   4096 Jan 25 17:59 .
argocd@argocd-repo-server-7dc6c547c5-xswgw:~$
```

The logic to create the cached copy of the Vault token in the `.avp` folder works fine, but notice it's owned by `argocd` user and group. 

I simulated OCP4 locally via
```
securityContext: 
   runAsUser: 1001
```

and ended up with the error described above. 

The weird thing is that I tried using the plugin from #41 which specifically adds error messages for this kind of situation, but I didn't see anything. Need to dig deeper there.

**Comment by jkayani** [comment]
Indeed, I confirm that by setting the SCC for the namespace ArgoCD runs in to `anyuid` so that the argocd-repo-server runs as `argocd`, the plugin version 0.2.0 works fine. Since I imagine most of Argo is written to run as it's specified non-root user, I don't think we should put effort in allowing the plugin to work with a random uid (`nonroot`, `restricted` SCC's).

**Comment by jkayani** [comment]
After a bit more debugging, all is clear:

v0.2.0 works only with argocd user since in that code,  the token is written to the file after login. Subsequent attempts to read the token to actually perform the work fail unless you are the right user. 

v0.2.2 works only with argocd user because it won't trigger an error trying to write the token to disk, which causes the error in this issue `yaml: mapping values are not allowed in this context` since Argo is now trying to interpret the string `error: somet error` as YAML

#41 has been amended to fix this, and to make a note on the README about the user requirements for getting the token caching to work. We can't emit "warning messages" to appear in Argo's UI unf, and we decided it would be better for the plugin to work via logging into Vault every time than not work because it couldn't cache the token.

**Comment by werne2j** [comment]
This fix has been published in [0.2.2](https://github.com/IBM/argocd-vault-plugin/releases/tag/v0.2.2)
